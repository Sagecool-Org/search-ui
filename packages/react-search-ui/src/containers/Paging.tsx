import React from "react";

import {
  Paging,
  PagingContainerProps,
  PagingViewProps
} from "@elastic/react-search-ui-views";
import { useSearch } from "../hooks";

export function PagingContainer({
  className,
  view,
  ...rest
}: PagingContainerProps) {
  const { current, resultsPerPage, totalPages, setCurrent } = useSearch();

  if (totalPages === 0) return null;

  const View: React.ComponentType<PagingViewProps> = view || Paging;

  const viewProps = {
    className,
    current,
    resultsPerPage,
    totalPages,
    onChange: setCurrent,
    ...rest
  };

  return <View {...viewProps} />;
}
export default PagingContainer;
