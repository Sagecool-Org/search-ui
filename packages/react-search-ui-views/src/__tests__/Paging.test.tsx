import React from "react";
import { render } from "@testing-library/react";
import Paging from "../Paging";

const params = {
  current: 1,
  onChange: () => ({}),
  resultsPerPage: 10,
  totalPages: 100
};

it("renders correctly", () => {
  const { container } = render(<Paging {...params} />);
  expect(container).toMatchSnapshot();
});

it("renders with className prop applied", () => {
  const customClassName = "test-class";
  const { container } = render(
    <Paging className={customClassName} {...params} />
  );
  expect(container.firstChild).toHaveClass(
    "rc-pagination",
    "sui-paging",
    "test-class"
  );
});
