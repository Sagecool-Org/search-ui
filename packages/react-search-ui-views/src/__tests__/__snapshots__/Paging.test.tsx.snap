// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <ul
    class="rc-pagination sui-paging"
  >
    <li
      aria-disabled="true"
      class="rc-pagination-prev rc-pagination-disabled"
      title="Previous Page"
    >
      <button
        aria-label="prev page"
        class="rc-pagination-item-link"
        disabled=""
        type="button"
      />
    </li>
    <li
      class="rc-pagination-item rc-pagination-item-1 rc-pagination-item-active"
      tabindex="0"
      title="1"
    >
      <a
        rel="nofollow"
      >
        1
      </a>
    </li>
    <li
      class="rc-pagination-item rc-pagination-item-2"
      tabindex="0"
      title="2"
    >
      <a
        rel="nofollow"
      >
        2
      </a>
    </li>
    <li
      class="rc-pagination-item rc-pagination-item-3"
      tabindex="0"
      title="3"
    >
      <a
        rel="nofollow"
      >
        3
      </a>
    </li>
    <li
      class="rc-pagination-item rc-pagination-item-4"
      tabindex="0"
      title="4"
    >
      <a
        rel="nofollow"
      >
        4
      </a>
    </li>
    <li
      class="rc-pagination-item rc-pagination-item-5 rc-pagination-item-before-jump-next"
      tabindex="0"
      title="5"
    >
      <a
        rel="nofollow"
      >
        5
      </a>
    </li>
    <li
      class="rc-pagination-jump-next"
      tabindex="0"
      title="Next 5 Pages"
    >
      <button
        aria-label="next page"
        class="rc-pagination-item-link"
        type="button"
      />
    </li>
    <li
      class="rc-pagination-item rc-pagination-item-100"
      tabindex="0"
      title="100"
    >
      <a
        rel="nofollow"
      >
        100
      </a>
    </li>
    <li
      aria-disabled="false"
      class="rc-pagination-next"
      tabindex="0"
      title="Next Page"
    >
      <button
        aria-label="next page"
        class="rc-pagination-item-link"
        type="button"
      />
    </li>
    <li
      class="rc-pagination-options"
    />
  </ul>
</div>
`;
